registration-processor/pre-processor/registration-processor-packet-classifier-stage/src/main/java/io/mosip/registration/processor/stages/packetclassifier/tagging/impl/MetaInfoTagGenerator.java

```diff
@@ -0,0 +1,160 @@
+        String operationsDataString = metaInfoMap.get(JsonConstant.OPERATIONSDATA);
+        if(operationsDataString == null)
+            throw new BaseCheckedException(
+                PlatformErrorMessages.RPR_PCM_OPERATIONS_DATA_ENTRY_NOT_AVAILABLE.getCode(), 
+                PlatformErrorMessages.RPR_PCM_OPERATIONS_DATA_ENTRY_NOT_AVAILABLE.getMessage());
+        JSONArray operationsDataJsonArray = new JSONArray(operationsDataString);
+        Map<String, String> operationsDataMap = getMapFromLabelValueArray(operationsDataJsonArray);
+        operationsDataTagLabels.forEach(tagLabel -> {
+            String tagLabelValue = operationsDataMap.get(tagLabel);
+            tags.put(operationsDataTagNamePrefix + tagLabel, tagLabelValue);
+        });
+        return tags;
+    }
+
+    private Map<String, String> generateTagsFromMetaData(Map<String, String> metaInfoMap) 
+            throws JSONException, JsonParseException, JsonMappingException, IOException, BaseCheckedException {
+        Map<String, String> tags = new HashMap<String, String>();
+        String metaDataString = metaInfoMap.get(JsonConstant.METADATA);
+        if(metaDataString == null)
+            throw new BaseCheckedException(
+                PlatformErrorMessages.RPR_PCM_META_DATA_ENTRY_NOT_AVAILABLE.getCode(), 
+                PlatformErrorMessages.RPR_PCM_META_DATA_ENTRY_NOT_AVAILABLE.getMessage());
+        JSONArray metaDataJsonArray = new JSONArray(metaDataString);
+        Map<String, String> metaDataMap = getMapFromLabelValueArray(metaDataJsonArray);
+        metaDataTagLabels.forEach(tagLabel -> {
+            String tagLabelValue = metaDataMap.get(tagLabel);
+            tags.put(metaDataTagNamePrefix + tagLabel, tagLabelValue);
+        });
+        return tags;
+    }
+
+    private Map<String, String> generateTagsFromCapturedRegisteredDevices(Map<String, String> metaInfoMap) 
+            throws JSONException, JsonParseException, JsonMappingException, IOException, BaseCheckedException {
+        Map<String, String> tags = new HashMap<String, String>();
+        String capturedRegisteredDevicesString = metaInfoMap.get(JsonConstant.CAPTUREDREGISTEREDDEVICES);
+        if(capturedRegisteredDevicesString == null)
+            throw new BaseCheckedException(
+                PlatformErrorMessages.RPR_PCM_CAPTURED_REGISTERED_DEVICES_ENTRY_NOT_AVAILABLE.getCode(), 
+                PlatformErrorMessages.RPR_PCM_CAPTURED_REGISTERED_DEVICES_ENTRY_NOT_AVAILABLE.getMessage());
+        JSONArray capturedRegisteredDevicesJsonArray = new JSONArray(capturedRegisteredDevicesString);
+        for(String deviceType : capturedRegisteredDeviceTypes) {
+            for(int i=0; i< capturedRegisteredDevicesJsonArray.length(); i++) {
+                JSONObject digitalId = capturedRegisteredDevicesJsonArray.getJSONObject(i)
+                    .getJSONObject(JsonConstant.DIGITALID);
+                if(digitalId.getString(JsonConstant.DIGITALIDTYPE).equals(deviceType))
```

Comment:
make, model, serial number?

Reply:
Should we also support taging config based on make and model and serial no? What should be taken as value of the tag? I have taken device type as config key and value as serial number

Reply:
As discussed, the tag value is changed to value concatinated string of make, model and serial number